src/Compiler/SyntaxTree/LexFilter.fs

```diff
@@ -1556,7 +1556,10 @@ type LexFilterImpl (
                 pushCtxtSeqBlock tokenTup NoAddBlockEnd
             | _ ->
                 // Queue a dummy token at this position to check if any closing rules apply
-                delayToken(pool.UseLocation(tokenTup, ODUMMY token))
+                if (match t2 with INTERP_STRING_END _ -> false | _ -> true) then
```

Comment:
Is this change specific to `INTERP_STRING_END`? Will it work differently if there're multiple inserts in an interpolated string?

Reply:
`INTERP_STRING_PART` is covered by previous case in this match and the problem is with this case only (when `INTERP_STRING_END` is "offside" in relation to `INTERP_STRING_BEGIN_PART` and too much is popped of the offside stack), so I suspect the answer is: yes.

I still need to understand this a little better and add some tests though.